stage0/run.go

```diff
@@ -491,6 +492,15 @@ func Run(cfg RunConfig, dir string, dataDir string) {
 		args = append(args, "--local-config="+cfg.LocalConfig)
 	}
 
+	s1v, err := getStage1InterfaceVersion(dir)
+	if err != nil {
+		log.FatalE("error determining stage1 interface version", err)
+	}
+
+	if cfg.Hostname != "" && s1v > 1 {
```

Comment:
I'd print a warning if we specified `--hostname`, but stage1 interface version is < 2.